aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,459 @@
+                assertEquals(
+                    AuthResetPasswordStep.CONFIRM_RESET_PASSWORD_WITH_CODE,
+                    result.getNextStep().getResetPasswordStep()
+                );
+                validateCodeDeliveryDetails(result.getNextStep().getCodeDeliveryDetails());
+            },
+            error -> fail()
+        );
+
+        verify(mobileClient, times(1))
+                .forgotPassword(any(String.class), any(Callback.class));
+    }
+
+    /**
+     * Tests that the confirmResetPassword method of the Auth wrapper of AWSMobileClient (AMC) calls
+     * AMC.confirmForgotPassword with the new password and confirmation code it received.
+     * Also ensures that in the onResult case, the success callback is triggered and in the onError case,
+     * the error call back receives an AuthException with the root cause attached.
+     */
+    @Test
+    public void confirmResetPassword() {
+        ForgotPasswordResult amcResult = new ForgotPasswordResult(ForgotPasswordState.DONE);
+
+        doAnswer(invocation -> {
+            assertEquals(newPassword, invocation.getArgument(0));
+            assertEquals(confirmationCode, invocation.getArgument(1));
+
+            Callback<ForgotPasswordResult> callback = invocation.getArgument(2);
+            callback.onResult(amcResult);
+            return null;
+        }).when(mobileClient)
+            .confirmForgotPassword(any(String.class), any(String.class), any(Callback.class));
+
+        authPlugin.confirmResetPassword(
+            newPassword,
+            confirmationCode,
+            () -> { },
+            error -> fail()
+        );
+
+        verify(mobileClient, times(1))
+                .confirmForgotPassword(any(String.class), any(String.class), any(Callback.class));
+    }
+
+    /**
+     * Tests that the getCurrentUser method of the Auth wrapper of AWSMobileClient (AMC) returns a new
+     * AWSCognitoAuthUser object containing the userId property in the plugin and the username from AMC.getUsername().
+     */
+    @Test
+    public void getCurrentUser() {
+        doAnswer(invocation -> username).when(mobileClient).getUsername();
+        AuthUser user = authPlugin.getCurrentUser();
+
+        assertEquals(userId, user.getUserId());
+        assertEquals(username, user.getUsername());
+    }
+
+    /**
+     * Tests that the getEscapeHatch method of the Auth wrapper of AWSMobileClient (AMC) returns the instance of
+     * AWSMobileClient held by the plugin.
+     */
+    @Test
+    public void getEscapeHatch() {
+        AWSMobileClient client = authPlugin.getEscapeHatch();
+        assertEquals(mobileClient, client);
+    }
+
+    /**
+     * Validate the sign up result is what was expected.
+     * @param result The received result
+     * @param targetStep The correct value for the next step (the only part of the response that varies in these tests)
+     */
+    private void validateSignUpResult(AuthSignUpResult result, AuthSignUpStep targetStep) {
```

Comment:
Instead of this, you can usually do like:
```
org.junit.Assert.assertEquals(highLevelExpectedObj, highLevelActualObj)
```

This works if the high level object has a sane `equals()` implementation.

Seems likely that `AWSMobileClient` (and its associated classes) may not.

If that's the case, the next best you can do is like:
```
private static void assertEquals(HighLevelType expected, HighLevelType actual) {
    // Assert field-by-field equality
}
```

Reply:
Yeah they aren't compatible and in this case I'm using default dummy values so while I could do it that second way, it'd just be less efficient code wise.